pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -127,70 +142,181 @@ func unpublishControllerCert(ctx *getconfigContext, key string) {
 func handleEveNodeCertModify(ctxArg interface{}, key string,
 	configArg interface{}) {
 
-	// XXX TBD
+	ctx := ctxArg.(*zedagentContext)
 	status := configArg.(types.EveNodeCert)
 	log.Infof("handleEveNodeCertModify for %s", status.Key())
+	triggerEveNodeCertEvent(ctx)
 	return
 }
 
 func handleEveNodeCertDelete(ctxArg interface{}, key string,
 	configArg interface{}) {
 
-	// XXX TBD
+	ctx := ctxArg.(*zedagentContext)
 	status := configArg.(types.EveNodeCert)
 	log.Infof("handleEveNodeCertDelete for %s", status.Key())
+	triggerEveNodeCertEvent(ctx)
 	return
 }
 
 // Run a task certificate post task, on change trigger
-func certsTask(ctx *zedagentContext, triggerCerts <-chan struct{}) {
-	iteration := 0
+func controllerCertsTask(ctx *zedagentContext, triggerCerts <-chan struct{}) {
+
 	log.Infoln("starting certs publish task")
-	publishCerts(ctx, iteration)
+	getCertsFromController(ctx)
 
 	// Run a periodic timer so we always update StillRunning
 	stillRunning := time.NewTicker(25 * time.Second)
-	agentlog.StillRunning(agentName+"certs", warningTime, errorTime)
+	agentlog.StillRunning(agentName+"ccerts", warningTime, errorTime)
 
 	for {
 		select {
 		case <-triggerCerts:
 			start := time.Now()
-			iteration++
-			publishCerts(ctx, iteration)
-			pubsub.CheckMaxTimeTopic(agentName+"certs", "publishCerts", start,
+			getCertsFromController(ctx)
+			pubsub.CheckMaxTimeTopic(agentName+"ccerts", "publishCerts", start,
 				warningTime, errorTime)
 
 		case <-stillRunning.C:
 		}
-		agentlog.StillRunning(agentName+"certs", warningTime, errorTime)
+		agentlog.StillRunning(agentName+"ccerts", warningTime, errorTime)
 	}
 }
 
 // prepare the certs list proto message
-func publishCerts(ctx *zedagentContext, iteration int) {
+func getCertsFromController(ctx *zedagentContext) bool {
+	certURL := zedcloud.URLPathString(serverNameAndPort,
+		zedcloudCtx.V2API, nilUUID, "certs")
+
+	// not V2API
+	if !zedcloud.UseV2API() {
+		return false
+	}
+
+	reqID := zcdevUUID.String()
+	zedcloud.RemoveDeferred("ccerts:" + reqID)
+
+	resp, contents, rtf, err := zedcloud.SendOnAllIntf(&zedcloudCtx,
+		certURL, 0, nil, 0, false)
+	if err != nil {
+		if rtf == types.SenderStatusRemTempFail {
+			log.Infof("getCertsFromController remoteTemporaryFailure: %s", err)
+		} else {
+			log.Errorf("getCertsFromController failed: %s", err)
+		}
+		zedcloud.SetDeferred("ccerts:"+reqID, nil, 0, certURL, zedcloudCtx, true)
+		return false
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK, http.StatusCreated, http.StatusNotModified:
+		log.Infof("getCertsFromController: status %s", resp.Status)
+	case http.StatusNotFound, http.StatusUnauthorized, http.StatusNotImplemented, http.StatusBadRequest:
+		log.Infof("getCertsFromController: server %s does not support V2 API", serverName)
```

Comment:
I think this info message is potentially misleading. Could get StatusNotFound etc for other reasons?
Would it be better to just have the default case printing of the status code?

Reply:
sure done.